openshift-kube-apiserver/admission/customresourcevalidation/dns/validate_dns_test.go

```diff
@@ -76,6 +113,33 @@ func TestSucceedValidateDNSSpec(t *testing.T) {
 				}},
 			},
 		},
+		"NetworkResolverType Upstream with TLS configured": {
+			UpstreamResolvers: operatorv1.UpstreamResolvers{
+				TransportConfig: operatorv1.DNSTransportConfig{
+					Transport: operatorv1.TLSTransport,
+				},
+				Upstreams: []operatorv1.Upstream{
+					{
+						Type:    operatorv1.NetworkResolverType,
+						Address: "1.1.1.1",
+					},
+				},
+			},
+		},
+		"Mixed Upstream types without TLS configured": {
```

Comment:
For completeness, should there also be coverage for TransportConfig.Transport == "Cleartext" and TransportConfig.TLS != nil (to test the default value of Transport when not returning early due to an empty TransportConfig)?

Reply:
Not a bad idea; fixed and pushing momentarily.